analysis/import/TokeiImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/tokeiimporter/strategy/TokeiTwelveStrategy.kt

```diff
@@ -0,0 +1,51 @@
+package de.maibornwolff.codecharta.importer.tokeiimporter.strategy
+
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+import com.google.gson.JsonObject
+import de.maibornwolff.codecharta.importer.tokeiimporter.analysisObject.AnalysisObjectTwelve
+import de.maibornwolff.codecharta.importer.tokeiimporter.analysisObject.Report
+import de.maibornwolff.codecharta.model.MutableNode
+import de.maibornwolff.codecharta.model.PathFactory
+import de.maibornwolff.codecharta.model.ProjectBuilder
+
+class TokeiTwelveStrategy(rootName: String, pathSeparator: String) : ImporterStrategy {
+    override var rootName = ""
+    override var pathSeparator = ""
+
+    init {
+        this.rootName = rootName
+        this.pathSeparator = pathSeparator
+    }
+
+    override fun buildCCJson(languageSummaries: JsonObject, projectBuilder: ProjectBuilder) {
+        val gson = Gson()
+        for (languageEntry in languageSummaries.entrySet()) {
+            val languageAnalysisObject = gson.fromJson(languageEntry.value, AnalysisObjectTwelve::class.java)
+            if (languageAnalysisObject.hasChildren()) {
+                for (report in languageAnalysisObject.reports) {
+                    addAsNode(report, projectBuilder)
+                }
+            }
+        }
+    }
+
+    override fun getLanguageSummaries(root: JsonElement): JsonObject {
+        return root.asJsonObject
+    }
+
+    private fun addAsNode(report: Report, projectBuilder: ProjectBuilder) {
+        val sanitizedName = report.name.removePrefix(rootName).replace(pathSeparator, "/")
+        val directory = sanitizedName.substringBeforeLast("/")
+        val fileName = sanitizedName.substringAfterLast("/")
+
+        val node = MutableNode(
+                fileName, attributes = mapOf(
+                "empty_lines" to report.stats.blanks,
+                "rloc" to report.stats.code,
+                "comment_lines" to report.stats.comments))
+
```

Comment:
The new parser seems to miss the lines of code? I guess they are not part of `report` but of `blob`? https://github.com/XAMPPRocky/tokei/blob/master/CHANGELOG.md#breaking-changes

Reply:
CodeStats? I didn't see that in the tokei.json.
To be fair, I couldn't find a proper API definition of their new json-format. :/

Reply:
I just checked the implementation and it seems like `loc` = `blanks` + `code` +`comments` (https://github.com/XAMPPRocky/tokei/blob/master/src/stats.rs#L26-L28)

Reply:
That makes sense. Let's calculate this metric by ourselfs then.

Reply:
I guess we continue to include the data in here? Or should we calculate it in UI in case it's not set?

Reply:
We don't calculate metrics in the visualization. The visualization is stupid and just shows whatever the cc.json provides. 
We can simply calculate loc from the values we got and add it to each node. Simple.